src/cmodel.c

```diff
@@ -1350,6 +1350,17 @@ static void CM_CalcChecksum(vfsfile_t *f, dheader_t *header, unsigned *checksum,
 		*checksum2 = map_checksum2;
 }
 
+static byte *CM_ReadLump(vfsfile_t *vf, lump_t *lump)
+{
+	byte *out;
+
+	VFS_SEEK(vf, lump->fileofs, SEEK_SET);
+	out = Hunk_TempAllocMore(lump->filelen);
```

Comment:
That hunk alloc may Host_Error resulting in vf handle leak, there is probably other cases.

Reply:
It probably not a Host_Error but Sys_Error which simply kills application, so probably not a leak, but code is still fragile.

Reply:
Yep, this is the assumption, and the reason for stacking all the CM_ReadLump's and CM_LoadBSPX/CM_BSPX_ReadLump and closing the file before the meat of the map loading. This mirrors the contract of FS_LoadFile which bails with Sys_Error if it failed to read. 

It feels like a more broad issue. Would be nice to have something similar to the Hunk area of clearing out memory to checkpoints also for file context structs to have a safe fallback.

A CM_LoadMap specific solution to this could be to reference the map file struct in a global, and if it's initialized next CM_LoadMap the old one could be closed.